Zend/zend_vm_def.h

```diff
@@ -4023,6 +4023,10 @@ ZEND_VM_HOT_HANDLER(130, ZEND_DO_UCALL, ANY, ANY, SPEC(RETVAL,OBSERVER))
 	zend_function *fbc = call->func;
 	zval *ret;
 
+	if (UNEXPECTED((fbc->common.fn_flags & ZEND_ACC_DEPRECATED) != 0)) {
+		zend_deprecated_function(fbc);
```

Comment:
You should probably check for `EG(exception)` after the calls to `zend_deprecated_function()` and abort the function call (`HANDLE_EXCEPTION();`). At least that seem to be the semantics for internal calls. https://3v4l.org/nFOWf

Not doing that will just abort on the first opcode that checks for exceptions anyway.

Reply:
It seems this part is still missing. Deprecated functions shouldn't be called if the error handler has thrown.

Reply:
Should be resolved now, please check whether my implementation is any good. I've had a few memory safety issues and I don't yet understand how the engine works.